lighthouse-core/lib/statistics.js

```diff
@@ -76,24 +51,37 @@ function getLogNormalScore({median, p10}, value) {
   // Required for the log-normal distribution.
   if (median <= 0) throw new Error('median must be greater than zero');
   if (p10 <= 0) throw new Error('p10 must be greater than zero');
-  // Not required, but if p10 > median, it flips around and becomes the p90 point.
+  // Not strictly required, but if p10 > median, it flips around and becomes the p90 point.
   if (p10 >= median) throw new Error('p10 must be less than the median');
 
   // Non-positive values aren't in the distribution, so always 1.
   if (value <= 0) return 1;
 
-  // Closest double to `erfc-1(2 * 1/10)`.
+  // Closest double to `erfc-1(1/5)`.
   const INVERSE_ERFC_ONE_FIFTH = 0.9061938024368232;
 
-  // Shape (σ) is `log(p10/median) / (sqrt(2)*erfc^-1(2 * 1/10))` and
+  // Shape (σ) is `|log(p10/median) / (sqrt(2)*erfc^-1(1/5))|` and
   // standardizedX is `1/2 erfc(log(value/median) / (sqrt(2)*σ))`, so simplify a bit.
-  const xLogRatio = Math.log(value / median);
-  const p10LogRatio = -Math.log(p10 / median); // negate to keep σ positive.
+  const xRatio = Math.max(Number.MIN_VALUE, value / median); // value and median are > 0, so is ratio.
+  const xLogRatio = Math.log(xRatio);
+  const p10Ratio = Math.max(Number.MIN_VALUE, p10 / median); // p10 and median are > 0, so is ratio.
+  const p10LogRatio = -Math.log(p10Ratio); // negate to keep σ positive.
```

Comment:
this is unchanged except the extra `Math.max()` calls that ensure that these values stay reasonable for some extreme cases where `value / median` or `p10 / median` underflow to 0 and NaN can result below.

Interestingly it appears that if a double `a` and `b` are greater than 0 and `a < b`, then `a / b < 1`, which is nice. I thought maybe rounding in the last place could sometimes bump it up to 1, but that appears to never happen (though I haven't come across a proof of this).

Reply:
> Interestingly it appears that if a double a and b are greater than 0 ...

How did you come to this conclusion? Once you go above any value over MAX_SAFE_INTEGER I find this to be not true.

```
for (let x = -100; x < 1000; x++) console.log(x, (Number.MAX_SAFE_INTEGER - 1 + x) / (Number.MAX_SAFE_INTEGER + x));
```

Reply:
Ah, I suppose I am failing to ensure that each integer has a unique repr. in the double format.

Reply:
right, strictly `a < b`